src/go/pt-k8s-pxc-recovery/recover/recover.go

```diff
@@ -0,0 +1,297 @@
+package recover
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/percona/percona-toolkit/src/go/pt-k8s-pxc-recovery/kubectl"
+)
+
+type Cluster struct {
+	Name          string
+	Size          int
+	MostRecentPod int
+	Namespace     string
+	ClusterImage  string
+}
+
+func (c *Cluster) SetClusterSize() error {
+	args := []string{
+		"get",
+		"pxc",
+		c.Name,
+		"-o",
+		"jsonpath='{.spec.pxc.size}'",
+	}
+	strSize, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return err
+	}
+	strSize = strings.Trim(strSize, "'")
+	c.Size, err = strconv.Atoi(strSize)
+	if err != nil {
+		return fmt.Errorf("error getting cluster size, %s", err.Error())
+	}
+	return nil
+}
+
+func (c *Cluster) GetClusterImage() error {
+	args := []string{
+		"get",
+		"pod",
+		c.Name + "-pxc-0",
+		"-o",
+		"jsonpath='{.spec.containers[0].image}'",
+	}
+	clusterImage, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return err
+	}
+	c.ClusterImage = strings.Trim(clusterImage, "'")
+	return nil
+}
+
+func (c *Cluster) getPods() ([]string, error) {
+	args := []string{
+		"get",
+		"pods",
+	}
+	out, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return []string{}, err
+	}
+	formatedOutput := strings.Split(out, "\n")
+	podNames := []string{}
+	for _, v := range formatedOutput {
+		podName := strings.Split(v, " ")[0]
+		if strings.Contains(podName, c.Name) && strings.Contains(podName, "pxc") {
+			podNames = append(podNames, podName)
+		}
+	}
+	return podNames, nil
+}
```

Comment:
I suggest using `kubectl get pods --no-headers -o custom-columns=":metadata.name" --selector=...`
With an appropriate selector you'll be able to get only pxc pods. Plus `-o custom-columns=` returns only pod names so no need in extra parsing. It seems much more reliable.